app/models/user_mailbox.rb

```diff
@@ -0,0 +1,44 @@
+class UserMailbox
+  attr_reader :user
+
+  def initialize(user)
+    raise ArgumentError.new("Must provide a user") if user == nil
```

Comment:
Yeah, this caught my eye too. You should always have a user, so rely on Ruby's failure message if the argument isn't supplied.